File: frontend/src/model/Note.ts
export class Note {
  id: number
  title: string
  description: string
  createdAt: string
  updatedAt: string
  is_deleted
  activeIndex!: boolean

  constructor (id: number, title: string, description: string, is_deleted: boolean) {
    this.id = id
    this.title = title
    this.description = description
    this.is_deleted = is_deleted
  }
}

File: backend/src/main/java/com/takenote/model/Note.java
package com.takenote.model;

import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.Column;
import javax.persistence.Entity;

@SuperBuilder
@AllArgsConstructor
@Entity
@Getter
@Setter
public class Note extends BaseEntity<Long> {

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    public Note() {
        this.isDeleted = false;
    }
}
